lib/node_modules/@stdlib/iter/cartesian-square/README.md

```diff
@@ -0,0 +1,133 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2024 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# iterCartesianSquare
+
+> Create an iterator which generates the Cartesian square of an input array-like object.
+
+<section class="intro">
+
+</section>
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var iterCartesianSquare = require( '@stdlib/iter/cartesian-square' );
+```
+
+#### iterCartesianSquare( x )
+
+Returns an iterator which generates the Cartesian Square of an input array-like object.
+
+```javascript
+var x = ['a', 'b', 'c'];
+var pair;
+var cartesianSquare = iterCartesianSquare( x, n );
+
+for ( pair of cartesianSquare ) {
+    console.log( pair );
+}
+
+```
+
+</section>
+
+<!-- /.usage -->
+
+<!-- Package usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="notes">
+
+## Notes
+
+The function expects only one argument x, an array-like object.
+The returned iterator will generate all possible combinations from the input array x.
+
+</section>
+
+<!-- /.notes -->
+
+<!-- Package usage examples. -->
+
+<section class="examples">
+
+## Examples
+
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+// Example : Generating Cartesian square of an array
+var x = [1, 2, 3];
+var cartesianSquare = iterCartesianSquare( x );
+var pair;
+
+for ( pair of cartesianSquare ) {
```

Comment:
We author everything in ES5. Perform manual iteration using a `while` loop. See other iterator packages.